lib/id3c/cli/command/geocode.py

```diff
@@ -0,0 +1,435 @@
+
+    cache = load_or_create_cache()
+
+    smartystreets_responses = pd.DataFrame(columns = ['lat','lng','canonicalized_address'])
+
+    for address in standardized_addresses:
+        response = get_response_from_cache_or_geocoding(address, cache)
+
+        smartystreets_responses = smartystreets_responses \
+            .append(response,ignore_index = True)
+
+    geocoded_addresses = address_file_df.join(smartystreets_responses)
+    save_cache(cache)
+
+    return geocoded_addresses
+
+
+def load_file_as_dataframe(filename: str) -> pd.DataFrame:
+    """
+    Given a *filename*, loads its data as a pandas DataFrame.
+
+    Raises a :class: `UnsupportedFileExtensionError` if the given *filename*
+    ends with an unsupported extension.
+    """
+    if not filename.endswith(SUPPORTED_EXTENSIONS):
+        raise UnsupportedFileExtensionError(dedent(f"""
+            Unsupported file extension for file «{filename}».
+            Please choose from one of the following file extensions:
+                {SUPPORTED_EXTENSIONS}
+            """
+        ))
+
+    if filename.endswith(('.csv', '.tsv')):
+        separator = '\t' if filename.endswith('.tsv') else ','
+        df = pd.read_csv(filename, sep=separator, dtype = str)
+    else:
+        df = pd.read_excel(filename, dtype = str)
+
+    return df
+
+
+def create_standardized_address_df(address_file_df: pd.DataFrame,
+                                   address_column_map: dict) -> pd.DataFrame:
+    """
+    Create a pandas DataFrame of addresses from an *address_file_df* using
+    the *address_column_map* to find and rename columns.
+
+    All address values are converted to uppercase and stripped of leading and
+    trailing whitespaces.
+    """
+    address_columns = list(filter(None, address_column_map.values()))
+
+    if not address_columns:
+        raise NoAddressColumnsFoundError(address_column_map)
+
+    addresses_df = pd.DataFrame()
+
+    for dst, src in address_column_map.items():
+        if src:
+            addresses_df[dst] = select_column(address_file_df, src)
+        else:
+            addresses_df[dst] = None
+
+    return addresses_df.applymap(lambda x: x.upper().strip() if x else None)
+
+
+def convert_df_to_dict_series(df: pd.DataFrame) -> pd.Series:
+    """
+    Given a pandas DataFrame *df*, convert each record into a dict of
+    {column -> value} and return list of records as a pandas Series.
+    """
+    return pd.Series(df.to_dict(orient='records'))
+
+
+def load_or_create_cache() -> TTLCache:
+    """
+    Tries to load a pickled cache from the filepath `cache.pickle`. If a cache
+    is not found at this location, creates a new one. Returns the cache.
+    """
+    try:
+        cache = pickle.load(open('cache.pickle', mode='rb'))
+    except FileNotFoundError:
+        LOG.warning("Couldn't find an existing cache file. Creating new cache.")
+        cache = TTLCache(maxsize=100000, ttl=CACHE_TTL)
```

Comment:
Should we increase the `maxsize`? AFAIK this number is completely arbitrary.

Reply:
Do we want `maxsize=float("inf")` like shown here https://github.com/tkem/cachetools/issues/73 ?

Reply:
Ah, yes, that looks great.